src/LoadingOverlay.tsx

```diff
@@ -1,40 +1,49 @@
-import React, { Component } from 'react'
-import PropTypes from 'prop-types'
+import React, { PureComponent } from 'react'
 import { CSSTransition } from 'react-transition-group'
 import { css, cx } from 'emotion'
 
 import Spinner from './components/Spinner'
 import STYLES from './styles'
+import {LoadingOverlayDefaultProps, LoadingOverLayProps, LoadingOverlayState, OverflowCSS} from './LoadingOverlayTypes'
+import {TransitionStatus} from 'react-transition-group/Transition';
 
-class LoadingOverlayWrapper extends Component {
-  constructor (props) {
+class LoadingOverlayWrapper extends PureComponent<LoadingOverLayProps, LoadingOverlayState> {
+  static defaultProps: LoadingOverlayDefaultProps = {
+    classNamePrefix: '_loading_overlay_',
+    fadeSpeed: 500,
+    styles: {}
+  }
+
+  wrapper = React.createRef<HTMLDivElement>();
+
+  constructor (props: LoadingOverLayProps) {
     super(props)
-    this.wrapper = React.createRef()
+
     this.state = { overflowCSS: {} }
   }
 
   componentDidMount () {
-    const wrapperStyle = window.getComputedStyle(this.wrapper.current)
-    const overflowCSS = ['overflow', 'overflowX', 'overflowY'].reduce((m, i) => {
+    const wrapperStyle: {[key: string]: any} = window.getComputedStyle(this.wrapper.current!)
+    const overflowCSS = ['overflow', 'overflowX', 'overflowY'].reduce((m: {[key: string]: string}, i) => {
       if (wrapperStyle[i] !== 'visible') m[i] = 'hidden'
       return m
     }, {})
     this.setState({ overflowCSS })
   }
 
-  componentDidUpdate (prevProps) {
+  componentDidUpdate (prevProps: LoadingOverLayProps) {
     const { active } = this.props
-    if (active) this.wrapper.current.scrollTop = 0
+    if (active && this.wrapper && this.wrapper.current) this.wrapper.current.scrollTop = 0
```

Comment:
here can use optional chaining `this.wrapper?.current`

Reply:
Changed it.